example/src/containers/MetaMaskConnector/MetaMaskConnector.tsx

```diff
@@ -11,14 +12,44 @@ export const MetaMaskConnector = () => {
     const installSnap = useCallback(async () => {
        const isInitiated = await installPolkadotSnap();
        if(!isInitiated) {
-           alert("Failed to install snap");
+          dispatch({type: MetamaskActions.HAS_INSTALL_FAILED, payload: true})
```

Comment:
You should modify code to receive error why installation failed. You can see in docs that error code 4001 means user rejected installation so you write "Please accept snap installation prompt"